### PR TITLE
feat(personality): `Quiet` personality

### PR DESCRIPTION
**Pull request by bene-dictator**
**Feature**
This PR addresses the feature described in #6052

## Summary
Creates a new personality, `quiet`, which does not send hails passively but will hail in the hail panel. Basically `mute` without the `(There is no response to your hail.)`

## Usage examples
See #6052

## Testing Done
I added this personality to all Merchant ships and went to Sol. After a minute on fast-forward, I only recieved hails from Republic ships. I then hailed a Merchant ship and I recieved a response.

## Performance Impact
Negligible?

### CODE REVIEW

**Review by Hurleveur — APPROVED** [review]
Looks good code wise
We can always add another personality for not sending hails in the hail panel if we need it (not suggesting it for this PR) and define mute as a mix of both if we need to later

File: source/Ship.cpp [hunk 1]
```diff
@@ -1519,7 +1519,7 @@ bool Ship::CanSendHail(const PlayerInfo &player, bool allowUntranslated) const
 		return false;
 
 	// Make sure this ship is able to send a hail.
```
Comment (RisingLeaf) [inline source/Ship.cpp:1521]: Update comment pls

**Review by RisingLeaf — APPROVED** [review]

File: source/Ship.cpp (hunk 1 above)
Comment (bene-dictator) [reply source/Ship.cpp:1521]: Something like this?
```suggestion
	// Make sure this ship is able to send a hail message, unless it has the quiet or mute personality.
```

Comment (RisingLeaf) [reply source/Ship.cpp:1521]: ye, also watch the length of the line, but I think it is fine

File: source/Ship.cpp [hunk 2]
```diff
@@ -1518,8 +1518,8 @@ bool Ship::CanSendHail(const PlayerInfo &player, bool allowUntranslated) const
 	if(!gov || IsYours())
 		return false;
 
-	// Make sure this ship is able to send a hail.
-	if(IsDisabled() || !Crew() || Cloaking() >= 1. || GetPersonality().IsMute())
+	// Make sure this ship is able to send a hail message, unless it has the quiet or mute personality.
```
Comment (Amazinite) [inline source/Ship.cpp:1521]: I don't see why this comment was updated. The new part doesn't really make sense, either. This if statement is still only making sure that this ship can send a hail; new personalities don't change that. "Unless" makes it sound like "If these personalities are here, then don't check if the ship is able to send a hail."

Comment (bene-dictator) [reply source/Ship.cpp:1521]: Revert it then?

Comment (Amazinite) [reply source/Ship.cpp:1521]: Yeah.

**Pull request merged**